irohad/pending_txs_storage/impl/pending_txs_storage_impl.cpp

```diff
@@ -60,13 +64,19 @@ PendingTransactionStorageImpl::getPendingTransactions(
 
   PendingTransactionStorage::Response response;
   response.all_transactions_size = account_batches.all_transactions_quantity;
-  auto remaining_space = page_size;
   while (account_batches.batches.end() != batch_iterator
-         and remaining_space >= batch_iterator->get()->transactions().size()) {
+         and (response.transactions.size()
+              + (*batch_iterator)->transactions().size())
+             <= page_size) {
     auto &txs = batch_iterator->get()->transactions();
-    response.transactions.insert(
-        response.transactions.end(), txs.begin(), txs.end());
-    remaining_space -= txs.size();
+    std::copy_if(txs.begin(),
+                 txs.end(),
+                 std::back_inserter(response.transactions),
+                 [&first_tx_time, &last_tx_time](auto const &tx) {
+                   auto const ts = tx->createdTime();
+                   return (!first_tx_time || ts > *first_tx_time)
+                       && (!last_tx_time || ts < *last_tx_time);
```

Comment:
I noticed that Your previous code was changed in MR: 
https://github.com/hyperledger/iroha/pull/1311/files?file-filters%5B%5D=.cpp :
![obraz](https://user-images.githubusercontent.com/15332594/129442690-e6e2dd22-0488-4da5-947c-3e0d8a02187f.png)
but don't worry - You don't have to correct tests,, because they were also changed.

I think for consistency it would be good to have `<=` instead of `<`.


@iceseer What do You think? Do You agree with my comment?

Reply:
I changed code proposed by @iceseer  because i did not see the PR you mentioned and thought <= was a mistake, but I will change it back to it